runtime/src/main/cpp/Arrays.cpp

```diff
@@ -417,7 +414,7 @@ void Kotlin_ByteArray_setFloatAt(KRef thiz, KInt index, KFloat value) {
 
 void Kotlin_ByteArray_setDoubleAt(KRef thiz, KInt index, KDouble value) {
   ArrayHeader* array = thiz->array();
-  if (static_cast<uint32_t>(index + 7) >= array->count_) {
+  if (index < 0 || static_cast<uint32_t>(index) + 7 >= array->count_) {
```

Comment:
Please ensure that we have tests for the code changed in this file.

Reply:
Good thing you noticed: we didn't have enough coverage indeed. I've added tests on out-of-bounds indexing and on out-of-bounds `addressOf` (on pinned objects). The former found problems with the `ByteArray::(get|set)(Type)At` implementation missing `index < 0`, and the latter also found a lack of `boundsCheck` [here](https://github.com/JetBrains/kotlin-native/pull/4319/files#diff-2fdb85cd4e883cd81e8bb6abdec07409R664)